src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -1816,7 +1816,19 @@ internal sealed override bool HasUnscopedRefAttribute
 
         internal sealed override bool UseUpdatedEscapeRules => ContainingModule.UseUpdatedEscapeRules;
 
-        internal sealed override bool IsCallerUnsafe => ContainingModule.UseUpdatedMemorySafetyRules && IsDeclaredRequiresUnsafe;
+        internal sealed override CallerUnsafeMode CallerUnsafeMode
+        {
+            get
+            {
+                if (ContainingModule.UseUpdatedMemorySafetyRules)
+                {
+                    return IsDeclaredRequiresUnsafe ? CallerUnsafeMode.Explicit : CallerUnsafeMode.None;
+                }
+
+                return this.HasParameterContainingPointerType() || ReturnType.ContainsPointerOrFunctionPointer()
```

Comment:
This feels like it could be somewhat expensive, and might be called often. I wonder if we should repurpose the bit in `_packedFlags` for just "does this thing require unsafe"; then, the body of this method would be:

```cs
(ContainingModule.UseUpdatedMemorySafetyRules, RequiresUnsafe) switch
{
    (_, false) => CallerUnsafeMode.None,
    (true, true) => CallerUnsafeMode.Explicit,
    (false, true) => CallerUnsafeMode.Explicit
};
```

Reply:
Or we could even go fancier with the bit packing, and simply say that 0b00 is unset, 0b01 is CallerUnsafeMode.None, 0b10 is CallerUnsafeMode.Implicit, and 0b11 is CallerUnsafeMode.Explicit.